tools/mo/unit_tests/mo/middle/StridedSliceShrinkNewAxisReplacer_test.py

```diff
@@ -0,0 +1,164 @@
+    **shaped_const_with_data('unsqueeze_axes', None)
+}
+
+pattern_edges = [
+    *connect('input', '0:strided_slice'),
+    *connect('begin', '1:strided_slice'),
+    *connect('end', '2:strided_slice'),
+    *connect('strided_slice', 'result')
+]
+
+
+class TestStridedSliceShrinkNewAxisReplacer(unittest.TestCase):
+
+    def test_replace_with_squeeze(self):
+        graph = build_graph(
+            nodes_attrs=nodes,
+            edges=pattern_edges,
+            update_attributes={
+                'strided_slice': {'shrink_axis_mask': [1, 0, 0, 0], 'new_axis_mask': [0, 0, 0, 0]},
+                'strided_slice_d': {'shape': [3, 5, 5]}
+            },
+            nodes_with_edges_only=True
+        )
+
+        ref_graph = build_graph(
+            nodes_attrs=nodes,
+            edges=[
+                *connect('input', '0:strided_slice'),
+                *connect('begin', '1:strided_slice'),
+                *connect('end', '2:strided_slice'),
+
+                *connect('strided_slice', '0:squeeze'),
+                *connect('squeeze_axes', '1:squeeze'),
+                *connect('squeeze', 'result'),
+            ],
+
+            update_attributes={
+                'squeeze_axes_d': {'value': [0]},
+                'squeeze_d': {'shape': [3, 5, 5]}
+            },
+            nodes_with_edges_only=True
+        )
+
+        ReplaceStridedSliceShrinkNewAxisWithSqueezeUnsqueeze().find_and_replace_pattern(graph)
+        (flag, resp) = compare_graphs(graph, ref_graph, 'result', check_op_attrs=True)
+        self.assertTrue(flag, resp)
+
+    def test_replace_with_unsqueeze(self):
+        graph = build_graph(
+            nodes_attrs=nodes,
+            edges=pattern_edges,
+            update_attributes={
+                'strided_slice': {'shrink_axis_mask': [0, 0, 0, 0], 'new_axis_mask': [1, 0, 0, 0]},
+                'strided_slice_d': {'shape': [1, 1, 3, 5, 5]}
+            },
+            nodes_with_edges_only=True
+        )
+
+        ref_graph = build_graph(
+            nodes_attrs=nodes,
+            edges=[
+                *connect('input', '0:strided_slice'),
+                *connect('begin', '1:strided_slice'),
+                *connect('end', '2:strided_slice'),
+                *connect('strided_slice', '0:unsqueeze'),
+                *connect('unsqueeze_axes', '1:unsqueeze'),
+                *connect('unsqueeze', 'result')
+            ],
+            update_attributes={
+                'unsqueeze_axes_d': {'value': [0]},
+                'unsqueeze_d': {'shape': [1, 1, 3, 5, 5]}
+            },
+            nodes_with_edges_only=True
+        )
+
+        ReplaceStridedSliceShrinkNewAxisWithSqueezeUnsqueeze().find_and_replace_pattern(graph)
+        graph = partial_infer(graph)
+        (flag, resp) = compare_graphs(graph, ref_graph, 'result', check_op_attrs=True)
+        self.assertTrue(flag, resp)
+
+    def test_replace_shrink_new_axis_with_squeeze_unsqueeze(self):
```

Comment:
Please add test when new_axis index is less then shrink

Reply:
I started to check these cases, and you are rights. Removing `new_axis_mask` if it's less than other masks we have to do some extra modifications.

Currently I am trying to solve this. Will update when it's done